request-manager/graph/resolver.go

```diff
@@ -41,41 +40,44 @@ func NewResolverFactory(jf job.Factory, specs map[string]*spec.Sequence, seqGrap
 
 func (f *resolverFactory) Make(req proto.Request) Resolver {
 	return &resolver{
-		Request:           req,
-		JobFactory:        f.jf,
-		AllSequences:      f.seqs,
-		SequenceGraphs:    f.seqGraphs,
-		IdGen:             f.idf.Make(),
+		request:        req,
+		jobFactory:     f.jf,
+		allSequences:   f.seqs,
+		sequenceGraphs: f.seqGraphs,
+		idGen:          f.idf.Make(),
 	}
 }
 
-// Creates a job chain for a single request.
+// Resolver builds a request graph for a specific request.
 type Resolver interface {
 	// Convert job args map to a list of proto.RequestArgs.
 	RequestArgs(jobArgs map[string]interface{}) ([]proto.RequestArg, error)
 
-	// Build the actual job chain. Returns an error if any error occurs.
-	BuildJobChain(jobArgs map[string]interface{}) (*proto.JobChain, error)
+	// Build the request graph. Returns an error if any error occurs.
+	BuildRequestGraph(jobArgs map[string]interface{}) (*Graph, error)
 }
 
-// Implements Resolver interface.
+// resolver implements the Resolver interface.
 type resolver struct {
-	Request           proto.Request              // the request spec this resolver can create job chain for
-	JobFactory        job.Factory                // factory to create nodes' jobs
-	AllSequences      map[string]*spec.Sequence  // sequence name --> sequence spec
-	SequenceGraphs    map[string]*Graph          // sequence name --> sequence graph
-	IdGen             id.Generator               // generates UIDs for jobs
+	request        proto.Request             // the request spec this resolver can create job chain for
+	jobFactory     job.Factory               // factory to create nodes' jobs
+	allSequences   map[string]*spec.Sequence // sequence name --> sequence spec
+	sequenceGraphs map[string]*Graph         // sequence name --> sequence graph
+	idGen          id.Generator              // generates UIDs for jobs
 }
 
+// RequestArgs takes user input args and returns them as a job args map, the form
+// expected by the Resolver later when building the request graph. Optional and
+// static args are filled in with their default values.
 func (o *resolver) RequestArgs(jobArgs map[string]interface{}) ([]proto.RequestArg, error) {
```

Comment:
Time to let go the past: let's `r *resolver`.  (`o` is because this part of the code used to be called Omega.)